src/main/java/com/toofifty/easyblastfurnace/methods/GoldBarMethod.java

```diff
@@ -38,32 +48,37 @@ private MethodStep checkPrerequisite(BlastFurnaceState state)
     @Override
     public MethodStep next(BlastFurnaceState state)
     {
-        MethodStep prerequisite = checkPrerequisite(state);
+        boolean hasGoldsmithEffect = state.getEquipment().hasGoldsmithEffect();
```

Comment:
I like the new method (can we also make one for ice gloves?), but there isn't really a need to store this and pass it around.